drivers/gpio/gpio_max22190.c

```diff
@@ -0,0 +1,651 @@
+		uint8_t max22190_24VME:1;
+		uint8_t max22190_24VLE:1;
+		uint8_t max22190_ALRMT1E:1;
+		uint8_t max22190_ALRMT2E:1;
+		uint8_t max22190_FAULT2E:1;
+		uint8_t max22190_PORE:1;
+		uint8_t max22190_CRCE:1; /* BIT 7 */
+	} reg_bits;
+} max22190_fault1_en;
+
+typedef union _max22190_fault2 {
+	uint8_t reg_raw;
+	struct {
+		uint8_t max22190_RFWBS:1; /* BIT 0 */
+		uint8_t max22190_RFWBO:1;
+		uint8_t max22190_RFDIS:1;
+		uint8_t max22190_RFDIO:1;
+		uint8_t max22190_OTSHDN:1;
+		uint8_t max22190_FAULT8CK:1;
+		uint8_t max22190_DUMMY:2; /* BIT 7 */
+	} reg_bits;
+} max22190_fault2;
+
+typedef union _max22190_fault2_en {
+	uint8_t reg_raw;
+	struct {
+		uint8_t max22190_RFWBSE:1; /* BIT 0 */
+		uint8_t max22190_RFWBOE:1;
+		uint8_t max22190_RFDISE:1;
+		uint8_t max22190_RFDIOE:1;
+		uint8_t max22190_OTSHDNE:1;
+		uint8_t max22190_FAULT8CKE:1;
+		uint8_t max22190_DUMMY:2; /* BIT 7 */
+	} reg_bits;
+} max22190_fault2_en;
+
+typedef union _max22190_cfg {
+	uint8_t reg_raw;
+	struct {
+		uint8_t max22190_DUMMY1:3; /* BIT 0 */
+		uint8_t max22190_24VF:1;
+		uint8_t max22190_CLRF:1;
+		uint8_t max22190_DUMMY2:2;
+		uint8_t max22190_REFDI_SH_EN:1; /* BIT 7 */
+	} reg_bits;
+} max22190_cfg;
+
+typedef union _max22190_filter {
+	uint8_t reg_raw;
+	struct {
+		uint8_t max22190_DELAY:3; /* BIT 0 */
+		uint8_t max22190_FBP:1;
+		uint8_t max22190_WBE:1;
+		uint8_t max22190_DUMMY:2; /* BIT 7 */
+	} reg_bits;
+} max22190_filter;
+
+enum max22190_delay {
+	MAX22190_DELAY_50US,
+	MAX22190_DELAY_100US,
+	MAX22190_DELAY_400US,
+	MAX22190_DELAY_800US,
+	MAX22190_DELAY_1800US,
+	MAX22190_DELAY_3200US,
+	MAX22190_DELAY_12800US,
+	MAX22190_DELAY_20000US
+};
+
+struct max22190_config {
+	struct spi_dt_spec spi;
+	struct gpio_dt_spec fault_gpio;
+	struct gpio_dt_spec ready_gpio;
+	struct gpio_dt_spec latch_gpio;
+	max22190_filter filter[8];
+	bool crc_en;
+	enum max22190_mode mode;
+	uint8_t pkt_size;
+};
+
+struct max22190_data {
+	struct gpio_driver_data common;
+	enum max22190_ch_state channels[MAX22190_CHANNELS];
+	enum max22190_ch_wb_state wb[MAX22190_CHANNELS];
+	max22190_cfg cfg;
+	max22190_fault1 fault1;
+	max22190_fault1_en fault1_en;
+	max22190_fault2 fault2;
+	max22190_fault2_en fault2_en;
+};
+
+/*
+ * @brief Compute the CRC5 value for MAX22190
+ * @param data - Data array to calculate CRC for.
+ * @return CRC result.
+ */
+static uint8_t max22190_crc(uint8_t *data)
+{
+	int length = 19;
+	uint8_t crc_step, tmp;
+	uint8_t crc_init = 0x7;
+	uint8_t crc_poly = 0x35;
+	int i;
+
+	/*
+	 * This is the C custom implementation of CRC function for MAX22190, and
+	 * can be found here:
+	 * https://www.analog.com/en/design-notes/guidelines-to-implement-crc-algorithm.html
+	 */
+	uint32_t datainput = (uint32_t)((data[0] << 16) + (data[1] << 8) + data[2]);
+
+	datainput = (datainput & 0xFFFFE0) + crc_init;
+
+	tmp = (uint8_t)((datainput & 0xFC0000) >> 18);
+
+	if ((tmp & 0x20) == 0x20)
+		crc_step = (uint8_t)(tmp ^ crc_poly);
+	else
+		crc_step = tmp;
```

Comment:
Add braces to every `if`, `else`, `do`, `while`, `for` and `switch` body, even for single-line code blocks
https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style

Reply:
Done